go/libraries/doltcore/str_to_noms.go

```diff
@@ -16,17 +16,17 @@ package doltcore
 
 import (
 	"errors"
+	"github.com/google/uuid"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/dolttypes"
 	"math"
 	"strconv"
 	"strings"
 
-	"github.com/google/uuid"
-
 	"github.com/liquidata-inc/dolt/go/store/types"
 )
 
-// StringToValue takes a string and a NomsKind and tries to convert the string to a noms Value.
-func StringToValue(s string, kind types.NomsKind) (types.Value, error) {
+// StringToValue takes a string, NomsKind, and DoltKind and tries to convert the string to a noms Value.
+func StringToValue(s string, kind types.NomsKind, doltKind dolttypes.DoltKind) (types.Value, error) {
```

Comment:
To keep these type signatures sane, you might want to introduce every type built on top of ByteArray as a new types.NomsKind constant.

Reply:
From what I can see, a `NomsKind` more or less maps with a `types.Value`. In a few places in the codebase, I'm seeing things like `uuid, ok := someParam.(types.UUID)`, which then works off of the assumption that you're dealing with a `UUID` without ever calling `Kind()`.

Reply:
you don't want to keep adding noms kinds for each.  Will make adding new kinds suck as much as before.  You'll have to go and fix a bunch of switch statements each time, etc.